js/reference.js

```diff
@@ -3097,45 +3110,52 @@
-                    for (var i = 0; i < this._data.length; i++) {
-
-                        // render template
-                        var value = module._renderTemplate(this._ref.display._markdownPattern, this._data[i], this._ref._table, this._ref._context);
-
-                        // If value is null or empty, return value on basis of `show_nulls`
-                        if (value === null || value.trim() === '') {
-                            value = module._getNullValue(this._ref._table, this._ref._context, [this._ref._table, this._ref._table.schema]);
-                        }
-
-                        // If final value is not null then push it in values array
-                        if (value !== null) {
-                            values.push(value);
-                        }
-                    }
-
-                    // Join the values array using the separator and prepend it with the prefix and append suffix to it.
-                    var pattern = this._ref.display._prefix + values.join(this._ref.display._separator) + this._ref.display._suffix;
-
-                    // Get the HTML generated using the markdown pattern generated above
-                    this._content =  module._formatUtils.printMarkdown(pattern);
-                } else {
                     this._content = null;
-                }
-            }
-
-            return this._content;
+                }else {
+                    var i, value, pattern, values = [];
+                    if (typeof this._ref.display._markdownPattern === 'string') {
+                       // Iterate over all data rows to compute the row values depending on the row_markdown_pattern.
+                       for (i = 0; i < this._data.length; i++) {
+                           // render template
+                           value = module._renderTemplate(this._ref.display._markdownPattern, this._data[i], this._ref._table, this._ref._context);
+
+                           // If value is null or empty, return value on basis of `show_nulls`
+                           if (value === null || value.trim() === '') {
+                               value = module._getNullValue(this._ref._table, this._ref._context, [this._ref._table, this._ref._table.schema]);
+                           }
+                           // If final value is not null then push it in values array
+                           if (value !== null) {
+                               values.push(value);
+                           }
+                       }
+
+                       // Join the values array using the separator and prepend it with the prefix and append suffix to it.
+                       pattern = this._ref.display._prefix + values.join(this._ref.display._separator) + this._ref.display._suffix;
+                   }else{
+                       if(this._ref.display._separator === undefined){
+                           this._ref.display._separator = "";
```

Comment:
I forgot about these cases that separator, prefix, and suffix might be undefined. Move these to the `get display` function. Set the initial value for them to the values that are defined there (`\n` for separator, and empty string for other variables).